framework/filters/AccessRule.php

```diff
@@ -108,7 +122,7 @@ class AccessRule extends Component
     public function allows($action, $user, $request)
     {
         if ($this->matchAction($action)
-            && $this->matchRole($user)
+            && $this->matchRole($user, $action)
```

Comment:
IMHO the point of any role is to allow or disallow an action.

But I can accept your worries about BC break, so if you like that I can remove this.